lib/cookie.c

```diff
@@ -1059,11 +1060,39 @@ Curl_cookie_add(struct Curl_easy *data,
               cllen = strlen(clist->spath);
 
             if(strncasecompare(clist->spath, co->spath, cllen)) {
+              infof(data, "cookie '%s' for domain '%s' dropped, would "
+                          "overlay an existing cookie", co->name, co->domain);
               freecookie(co);
               return NULL;
             }
           }
-          else if(strcasecompare(clist->spath, co->spath))
+        }
+      }
+    }
+    lastc = clist;
+    clist = clist->next;
+  }
+
+  clist = c->cookies[myhash];
+  replace_old = FALSE;
+  while(clist) {
```

Comment:
Traversing clist twice adds a computational cost, might be unavoidable but we should probably at the least add a comment on why this is done.

Reply:
I'll see if I can work it down to a single loop

Reply:
Just needed a little massaging. Thanks for highlighting this.